packages/style-engine/class-wp-style-engine.php

```diff
@@ -293,6 +288,158 @@ protected static function is_valid_style_value( $style_value ) {
+		}
+		// Register the hook callback to render stored styles to the page.
+		static::render_styles( array( __CLASS__, 'process_and_enqueue_stored_styles' ) );
+	}
+
+	/**
+	 * Utility method to retrieve the main instance of the class.
+	 *
+	 * The instance will be created if it does not exist yet.
+	 *
+	 * @return WP_Style_Engine The main instance.
+	 */
+	public static function get_instance() {
+		if ( null === static::$instance ) {
+			static::$instance = new static();
+		}
+
+		return static::$instance;
+	}
+
+	/**
+	 * Stores a CSS rule using the provide CSS selector and CSS declarations.
+	 *
+	 * @param string $css_selector     When a selector is passed, the function will return a full CSS rule `$selector { ...rules }`, otherwise a concatenated string of properties and values.
+	 * @param array  $css_declarations An array of parsed CSS property => CSS value pairs.
+	 * @param string $store_key        A valid key corresponding to an existing store in static::$stores.
+	 *
+	 * @return void.
+	 */
+	public static function store_css_rule( $css_selector, $css_declarations, $store_key ) {
+		if ( ! $css_selector || ! isset( static::$stores[ $store_key ] ) ) {
+			return;
+		}
+		$css_declarations = new WP_Style_Engine_CSS_Declarations( $css_declarations );
+		$stored_css_rule  = static::$stores[ $store_key ]->add_rule( $css_selector );
+		$stored_css_rule->add_declarations( $css_declarations );
+	}
+
+	/**
+	 * Returns a store by store key.
+	 *
+	 * @param string $store_key A valid key corresponding to an existing store in static::$stores.
+	 *
+	 * @return WP_Style_Engine_CSS_Rules_Store|null The store, if found, otherwise `null`.
+	 */
+	public static function get_store( $store_key ) {
+		if ( ! isset( static::$stores[ $store_key ] ) ) {
+			return null;
+		}
+		return static::$stores[ $store_key ];
+	}
+
+	/**
+	 * Taken from gutenberg_enqueue_block_support_styles()
+	 *
+	 * This function takes care of adding inline styles
+	 * in the proper place, depending on the theme in use.
+	 *
+	 * For block themes, it's loaded in the head.
+	 * For classic ones, it's loaded in the body
+	 * because the wp_head action  happens before
+	 * the render_block.
+	 *
+	 * @param callable $callable A user-defined callback function for a WordPress hook.
+	 * @param int      $priority To set the priority for the add_action.
+	 *
+	 * @see gutenberg_enqueue_block_support_styles()
+	 */
+	protected static function render_styles( $callable, $priority = 10 ) {
+		if ( ! $callable ) {
+			return;
+		}
+		$action_hook_name = 'wp_footer';
+		if ( wp_is_block_theme() ) {
+			$action_hook_name = 'wp_head';
+		}
+		add_action( 'wp_enqueue_scripts', $callable );
+		add_action(
+			$action_hook_name,
+			$callable,
+			$priority
+		);
+	}
+
+	/**
+	 * Fetches, processes and compiles stored styles, then renders them to the page.
+	 */
+	public static function process_and_enqueue_stored_styles() {
```

Comment:
The enqueue and WP hook logic could well live in Gutenberg itself, e.g., `script-loader.php` with the style engine only taking care of `parse + store + compile`